File: host-backend-integration-api-proxy-security/application-host-express/src/ExampleAPIProxyWithSecuritySetup.ts
import { IncomingMessage } from 'http';
import {OpenMicrofrontendsExampleAPIProxyWithSecurityBaseSetup} from './_generated/microfrontendHostIntegrations';

export default class ExampleAPIProxyWithSecuritySetup implements OpenMicrofrontendsExampleAPIProxyWithSecurityBaseSetup {
    get microfrontendBaseUrl() {
        return 'http://localhost:7830'
    };

    async getUser(req: IncomingMessage)  {
        // TODO
        return null;
    }

    async apiProxyRequestBffGetSecurityHeaders(req: IncomingMessage): Promise<Record<string, string>> {
        return {
            'x-api-key': '123456',
        };
    }

    get apiProxyChuckNorrisJokeUrl() {
        return 'https://api.chucknorris.io/jokes/random';
    };

    async userPermissionsRequestGetSecurityHeaders(req: IncomingMessage): Promise<Record<string, string>> {
        return {
          'x-api-key': '123456',
        };
    }

    async buildTimestampOrVersionCachePut(tsOrVersion: string): Promise<void> {
        // TODO
    }

    async buildTimestampOrVersionCacheGet(): Promise<string | null> {
        // TODO
        return null;
    }
}

File: host-backend-integration-api-proxy-security/host-application-spring-boot/src/main/java/org/openmicrofrontends/demo/ExampleAPIProxyWithSecuritySetup.java
package org.openmicrofrontends.demo;

import java.util.Map;

import jakarta.servlet.http.HttpServletRequest;
import org.openmicrofrontends.demo._generated.OpenMicrofrontendHostIntegrations;

public class ExampleAPIProxyWithSecuritySetup implements
        OpenMicrofrontendHostIntegrations.OpenMicrofrontendsExampleAPIProxyWithSecurityHostIntegrationFilter.MicrofrontendBaseSetup {

    @Override
    public String getMicrofrontendBaseUrl() {
        return "http://localhost:7830";
    }

    @Override
    public OpenMicrofrontendHostIntegrations.User getUser(HttpServletRequest req) {
        // TODO
        return null;
    }

    @Override
    public Map<String, String> getApiProxyRequestBffGetSecurityHeaders(HttpServletRequest req) {
        return Map.of("x-api-key", "123456");
    }

    @Override
    public String getApiProxyChuckNorrisJokeUrl() {
        return "https://api.chucknorris.io/jokes/random";
    }

    @Override
    public Map<String, String> getUserPermissionsRequestSecurityHeaders(HttpServletRequest req) {
        return Map.of("x-api-key", "123456");
    }

    @Override
    public void buildTimestampOrVersionCachePut(String tsOrVersion) {
        // TODO
    }

    @Override
    public String buildTimestampOrVersionCacheGet() {
        // TODO
        return null;
    }
}
